packages/types/vitest.config.ts

```diff
@@ -0,0 +1,9 @@
+import { defineConfig } from 'vitest/config'
+
+export default defineConfig({
+  test: {
+    typecheck: {
+      enabled: true,
```

Comment:
the existing vitest type assertions in `packages/types` were also not doing anything because of this